### PR TITLE
[FIX] Disallow empty string/whitespace-only `pheno --name` values

### LINKED ISSUE #418: `pheno` command allows empty strings/blankspace for `--name` option
- [ ] should also update description

### PR DESCRIPTION
**Pull request by alyssadai**
- Fixes #418 

Changes proposed in this pull request:

- Add custom validation to check that the value of `--name` is not `""` or whitespace
- Clarify in help text that `--name` determines the dataset display name in Neurobagel query results

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Validate the dataset name passed to the `pheno` command to ensure it is not an empty string or whitespace. Update the help text to clarify that the name is used as the dataset display name in Neurobagel query results.

Bug Fixes:
- Prevent an error where an empty string or whitespace-only value could be provided for the dataset name in the `pheno` command.

Tests:
- Add tests to verify that an empty string or whitespace-only value for the `--name` argument results in an error.

### CODE REVIEW

File: bagel/cli.py [hunk 1]
```diff
@@ -54,9 +54,11 @@ def pheno(
         ...,
         "--name",
         "-n",
-        help="A descriptive name for the dataset the input belongs to. "
-        "This name is expected to match the name field in the BIDS dataset_description.json file. "
-        'Should be enclosed in quotes, e.g.: --name "my dataset name"',
+        callback=pheno_utils.validate_dataset_name,
+        help="The full name of the dataset. "
+        "This name will be displayed when the dataset matches a Neurobagel query. "
```
Comment (surchs) [inline bagel/cli.py:59]:
```suggestion
        "This name will be displayed when users discover the dataset in a Neurobagel query. "
```

File: bagel/cli.py [hunk 2]
```diff
@@ -54,9 +54,11 @@ def pheno(
...
+        "For a dataset with BIDS data, should ideally match the dataset_description.json 'name' field. "
```
Comment (surchs) [inline bagel/cli.py:60]:
```suggestion
        "For a dataset with BIDS data, the name should ideally match the dataset_description.json 'name' field. "
```

**Review by surchs — APPROVED** [review]
Very cool, thanks @alyssadai :tada: 

:cook:
